File: sdks/go/pkg/models/operations/putfilesidversionsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutFilesIDVersionsIDPathParams struct {
	FileID        string `pathParam:"style=simple,explode=false,name=file_id"`
	FileVersionID string `pathParam:"style=simple,explode=false,name=file_version_id"`
}

// PutFilesIDVersionsIDRequestBody
// The file version to be
// restored
type PutFilesIDVersionsIDRequestBody struct {
	TrashedAt *string `json:"trashed_at,omitempty"`
}

type PutFilesIDVersionsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams PutFilesIDVersionsIDPathParams
	Request    *PutFilesIDVersionsIDRequestBody `request:"mediaType=application/json"`
}

type PutFilesIDVersionsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	FileVersion *shared.FileVersion
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/putfilesidversionsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutFilesIdVersionsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_version_id" })
  fileVersionId: string;
}


// PutFilesIdVersionsIdRequestBody
/** 
 * The file version to be
 * restored
**/
export class PutFilesIdVersionsIdRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=trashed_at" })
  trashedAt?: string;
}


export class PutFilesIdVersionsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutFilesIdVersionsIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutFilesIdVersionsIdRequestBody;
}


export class PutFilesIdVersionsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileVersion?: shared.FileVersion;

  @SpeakeasyMetadata()
  statusCode: number;
}
